src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java

```diff
@@ -261,9 +338,8 @@ private void applyArtifactInNormalFilter(final M2FiltersArgumentCollection MTFAC
         // the above filter misses artifacts whose support in the normal consists entirely of low base quality reads
         // Since a lot of low-BQ reads is itself evidence of an artifact, we filter these by hand via an estimated LOD
         // that uses the average base quality of *ref* reads in the normal
-        final int normalMedianRefBaseQuality = GATKProtectedVariantContextUtils.getAttributeAsIntArray(
-                normalGenotype, BaseQuality.KEY, () -> new int[] {IMPUTED_NORMAL_BASE_QUALITY}, IMPUTED_NORMAL_BASE_QUALITY)[0];
-        final double normalPValue = 1 - new BinomialDistribution(null, normalDepth, QualityUtils.qualToErrorProb(normalMedianRefBaseQuality))
+        final int medianRefBaseQuality = vc.getAttributeAsIntList(BaseQuality.KEY, IMPUTED_NORMAL_BASE_QUALITY).get(0);
```

Comment:
I like your abstract PerAlleleAnnotation method that builds the header lines, but I'd rather that BaseQuality.KEY referred to a value in GATKVCFConstants.

Reply:
done